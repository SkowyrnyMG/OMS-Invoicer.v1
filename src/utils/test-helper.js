import React from 'react';
import { Router } from 'react-router-dom';
import { render, fireEvent } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { createMemoryHistory } from 'history';
import { theme } from 'themes/theme';

export const TestThemeProvider = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

const renderWithRouter = (ui, { route = '/', ...renderOptions } = {}) => {
  const history = createMemoryHistory({ initialEntries: [route] });
  const utils = render(<Router history={history}>{ui}</Router>, renderOptions);
  return {
    ...utils,
    history,
  };
};

const leftClick = { button: 0 };

export { leftClick, renderWithRouter, fireEvent };
