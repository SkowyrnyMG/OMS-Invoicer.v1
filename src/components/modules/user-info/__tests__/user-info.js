import React from 'react';

import { snapShotTest } from 'utils/tests/test-helper';
import UserInfo from '../user-info';

describe('UserInfo', () => {
  test('should render in the document with correct styles', () => {
    snapShotTest(<UserInfo>testuser</UserInfo>);
  });
});
