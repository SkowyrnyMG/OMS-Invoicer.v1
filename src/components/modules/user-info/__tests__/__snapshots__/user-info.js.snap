// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo should render in the document with correct styles 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
}

.c2 {
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  font-size: 14px;
  color: #446DF6;
}

.c1 {
  height: 1.4rem;
  fill: #446DF6;
}

<div
  class="c0"
>
  <svg
    class="c1"
  >
    user-icon.svg
  </svg>
  <span
    class="c2"
  >
    testuser
  </span>
</div>
`;
